### PR TITLE
Explorer rewrite

### PR DESCRIPTION
**Pull request by zorkow**
Rewrite of the explorer that fixes a number longstanding issues and introduces the new auto voicing feature.

## Explore Pool

All explorers for a `MathItem` are now held in a dedicated `ExplorePool` object. Consequently, all explorers in one pool use only one highlighter, which is administered by the pool. Consequently there should no longer be interaction of highlighting by multiple explorers. 
In addition, the pool offers a second highlighter used for synchronised highlighting during auto voicing (see below).

The pool also ensures clean re-attaching and restarting of explorers on re-rendering. In that case all the explorers are re-initialised, while the pool stays the same object. As before, upon re-rendering we try to continue exploration at the sam position in the term as before (e.g., when a sub-term  was collapsed on that sub-term). This works, well in the case of our collapse actions, however in the case of general toggle actions, this can potentially end in the wrong element.

## Autovoicing

This is implemented in the new `SpeechRegion` class that takes care of extracting the SSML returned from SRE, putting together the `utterances` for the `speechSynthesis`  API and the synchronised highlighting. This is superimposed in red over the normal level highlighting. I was originally thinking about using a frame but my CSS attempts looked ugly.

Note, that autovoicing is currently hidden behind a flag. In the future we should have a hotkey for switching it on.
Note also, that at the moment `SpeechRegion` this is still a live region. 

## Exploration method

Replaces aria role `application` with `tree`. There is relatively little else done, as we can simply reuse the `ARIA` structure that SRE already produces given the `structure` feature.  I am still in the process of testing this in more detail, so we might need another PR, once that is fully worked out.

## SRE version

This currently needs SRE v4.1.0-beta.1 to work.

### CODE REVIEW

File: ts/a11y/explorer/Region.ts [hunk 1]
```diff
@@ -367,6 +367,273 @@ export class LiveRegion extends StringRegion {
...
+    {color: 'red'}, {color: 'black'},
+    {renderer: this.document.outputJax.name, browser: 'v3'}
+  );
+
+  /**
+   * @override
+   */
+  public Clear(): void {
+    super.Clear();
+  }
```
Comment (dpvc) [inline ts/a11y/explorer/Region.ts:424]: I'm not sure why this is being overridden since it only does the super action anyway.

**Review by dpvc — APPROVED** [review]
This looks like a good update.  I must admit that I didn't follow everything entirely, but don't see anything to complain about, so I trust you on it.

File: ts/a11y/explorer.ts [hunk 2]
```diff
@@ -31,11 +31,7 @@ import {OptionList, expandable} from '../util/Options.js';
 import {SerializedMmlVisitor} from '../core/MmlTree/SerializedMmlVisitor.js';
 import {MJContextMenu} from '../ui/menu/MJContextMenu.js';
 
-import {Explorer} from './explorer/Explorer.js';
-import * as ke from './explorer/KeyExplorer.js';
-import * as me from './explorer/MouseExplorer.js';
-import {TreeColorer, FlameColorer} from './explorer/TreeExplorer.js';
-import {LiveRegion, ToolTip, HoverRegion} from './explorer/Region.js';
+import { ExplorerPool, RegionPool } from './explorer/ExplorerPool.js';
```
[the remark this answers is not in the document]
Comment (zorkow) [reply ts/a11y/explorer.ts]: A habit from SRE which uses `prettier` that adds these spaces automatically.

File: ts/a11y/explorer/Explorer.ts [hunk 3]
```diff
@@ -44,6 +46,8 @@ export interface Explorer {
    */
   stoppable: boolean;
 
+  pool: ExplorerPool;
```
[the remark this answers is not in the document]
Comment (zorkow) [reply ts/a11y/explorer/Explorer.ts:49]: dito

File: ts/a11y/explorer/Region.ts (hunk 1 above)
Comment (zorkow) [reply ts/a11y/explorer/Region.ts:424]: Probably did something during experimental phase. Removed now.

**Pull request merged**